### PR TITLE
feat: entry functions to lock more and unlock composable vetoken

### PR DESCRIPTION
**Pull request by wjw12**

### CODE REVIEW

**Review by 0xbe1 — APPROVED** [review]
lgtm

File: sources/scripts.move [hunk 1]
```diff
@@ -44,14 +42,32 @@ module vetoken::scripts {
         let coin_b = coin::withdraw<LockCoinB>(account, amount_b);
         composable_vetoken::lock(account, coin_a, coin_b, epochs);
     }
+    
+    public entry fun increase_lock_amount_and_duration_composed<LockCoinA, LockCoinB>(account: &signer, amount_a: u64, amount_b: u64, increment_epochs: u64) {
+        let account_addr = signer::address_of(account);
+        if (coin::is_account_registered<LockCoinA>(account_addr)) {
```
Comment (0xbe1) [inline sources/scripts.move:48]: should it be `vetoken::is_account_registered`?

File: sources/composable_vetoken.move [hunk 2]
```diff
@@ -144,6 +144,12 @@ module vetoken::composable_vetoken {
         (total_supply_a * weight_percent_coin_a, total_supply_b * weight_percent_coin_b)
     }
 
+    #[view]
+    /// Returns true if the account has either token locked
+    public fun has_lock<CoinType>(account_addr: address): bool {
```
Comment (wjw12) [inline sources/composable_vetoken.move:149]: adding this to help Dapp decide whether to use `lock_composed` or `increase_lock_amount_and_duration_composed`

Comment (0xbe1) [reply sources/composable_vetoken.move:149]: prefer `either_locked` that's more accurate

File: sources/scripts.move [hunk 3]
```diff
@@ -44,14 +42,54 @@ module vetoken::scripts {
...
         composable_vetoken::lock(account, coin_a, coin_b, epochs);
     }
+    
+    /// (1) If neither A nor B is registered, do nothing (users should call `lock_composed` instead)
+    /// (2) If both A and B are registered, this function assumes two tokens have the same lock duration (which is true if user only interacts with Dapp) 
+    /// and increases the lock amount/duration of both tokens
+    /// (3) If only one of A and B is registered, this function increases the lock amount/duration of the registered token, and create a new lock for the other token,
+    /// with the same lock duration as the registered token
+    /// If any of the two tokens is unlockable, this function will abort in the first `increase_lock_amount_and_duration` call of that token. We don't handle such an edge case
+    public entry fun increase_lock_amount_and_duration_composed<LockCoinA, LockCoinB>(account: &signer, amount_a: u64, amount_b: u64, increment_epochs: u64) {
```
Comment (0xbe1) [inline sources/scripts.move:52]: i believe we need unit tests for it

**Comment by wjw12** [comment]
cancelling

**Pull request closed without merging**